Detectors/TOF/base/src/Geo.cxx

```diff
@@ -19,3 +20,478 @@ constexpr Float_t Geo::HEIGHTS[NPLATES][NMAXNSTRIP];
 constexpr Float_t Geo::DISTANCES[NPLATES][NMAXNSTRIP];
 constexpr Bool_t Geo::FEAWITHMASKS[18];
 constexpr Float_t Geo::ROOF2PARAMETERS[3];
+
+void Geo::GetDetID(Float_t* pos, Int_t* det)
+{
```

Comment:
Hi @noferini , this method looks quite inefficient, since the same quantities (e.g. GetSector) are calculated over and over in different Get.. methods called by it. 
1st of all, could you consider filling instead of array det a struct defined within the class like
```
struct Geo::DetData {
short sector;
short plate;
short strip
short padZ;
short padX;
};  
...

void Geo::GetDetID(Float_t* pos, DetData &data)
```
then avoid recalculating params which where already calculated, more comments below